libraries/rush-lib/src/logic/pnpm/PnpmfileShim.ts

```diff
@@ -40,7 +49,7 @@ function init(context: IPnpmfileContext | any): IPnpmfileContext {
     if (!context.pnpmfileShimSettings) {
       context.pnpmfileShimSettings = __non_webpack_require__('./pnpmfileSettings.json');
```

Comment:
Were the installs running in the same process? If not I don't see how there could be contamination of in-memory state, and if so, then we need to purge the `require.cache` entry corresponding to `./pnpmfileSettings.json` to ensure that we get the latest.